src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,71 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private static final Map<Class<?>, Class<?>> interfaceImplementation =
+            Map.of(FileReaderService.class, FileReaderServiceImpl.class,
+                    ProductParser.class, ProductParserImpl.class,
+                    ProductService.class, ProductServiceImpl.class);
+    private static final Map<Class<?>, Object> instances = new HashMap<>();
 
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        Object classImplementationInstance = null;
+        Class<?> classImpl = findImplementationClass(interfaceClazz);
+        if (!classImpl.isAnnotationPresent(Component.class)) {
+            throw new RuntimeException("Can't initialize object without Component annotation");
+        }
+        Field[] declaredFields = interfaceClazz.getDeclaredFields();
+        for (Field field : declaredFields) {
+            if (field.isAnnotationPresent(Inject.class)) {
+                Object fieldInstance = getInstance(field.getType());
+                classImplementationInstance = createNewInstance(classImpl);
+                try {
+                    field.setAccessible(true);
+                    field.set(classImplementationInstance, fieldInstance);
+                } catch (IllegalAccessException e) {
+                    throw new RuntimeException("Can't initialize field value. "
+                            + "Class: " + classImpl.getName() + "Field: " + field.getName());
```

Comment:
don't forget to put a stack trace from the exception 
```suggestion
                    throw new RuntimeException("Can't initialize field value. "
                            + "Class: " + classImpl.getName() + "Field: " + field.getName(), e);
```